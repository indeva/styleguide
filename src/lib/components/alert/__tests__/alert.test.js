import React from 'react';
import renderer from 'react-test-renderer';
import Alert from '..';

describe('Alert', () => {
  describe('snapshot tests', () => {
    const defaultProps = {
      visible: true,
      onCloseClick: () => {},
      showClose: false,
      type: 'success',
    };

    it('should render danger alert properly', () => {
      const component = renderer.create(<Alert {...defaultProps} type="danger">Test</Alert>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render warning alert properly', () => {
      const component = renderer.create(<Alert {...defaultProps} type="warning">Test</Alert>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render success alert properly', () => {
      const component = renderer.create(<Alert {...defaultProps} type="success">Test</Alert>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render success alert properly', () => {
      const component = renderer.create(<Alert {...defaultProps} type="success">Test</Alert>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render with close button when showClose is true', () => {
      const component = renderer.create(<Alert {...defaultProps} showClose>Test</Alert>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render with className', () => {
      const component = renderer.create(<Alert {...defaultProps} className="test">Test</Alert>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render null when visible is false', () => {
      const component = renderer.create(<Alert {...defaultProps} visible={false}>Test</Alert>);
      const tree = component.toJSON();
      expect(tree).toBeNull();
    });
  });
});
