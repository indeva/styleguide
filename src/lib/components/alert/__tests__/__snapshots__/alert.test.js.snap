// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert snapshot tests should render danger alert properly 1`] = `
<div
  className="alert alert--danger"
>
  <div
    className="alert__body"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times-circle fa-w-16 alert__icon alert__icon--danger"
      data-icon="times-circle"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm101.8-262.2L295.6 256l62.2 62.2c4.7 4.7 4.7 12.3 0 17l-22.6 22.6c-4.7 4.7-12.3 4.7-17 0L256 295.6l-62.2 62.2c-4.7 4.7-12.3 4.7-17 0l-22.6-22.6c-4.7-4.7-4.7-12.3 0-17l62.2-62.2-62.2-62.2c-4.7-4.7-4.7-12.3 0-17l22.6-22.6c4.7-4.7 12.3-4.7 17 0l62.2 62.2 62.2-62.2c4.7-4.7 12.3-4.7 17 0l22.6 22.6c4.7 4.7 4.7 12.3 0 17z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="alert__content"
    >
      Test
    </span>
  </div>
</div>
`;

exports[`Alert snapshot tests should render success alert properly 1`] = `
<div
  className="alert alert--success"
>
  <div
    className="alert__body"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-check-circle fa-w-16 alert__icon alert__icon--success"
      data-icon="check-circle"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="alert__content"
    >
      Test
    </span>
  </div>
</div>
`;

exports[`Alert snapshot tests should render success alert properly 2`] = `
<div
  className="alert alert--success"
>
  <div
    className="alert__body"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-check-circle fa-w-16 alert__icon alert__icon--success"
      data-icon="check-circle"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="alert__content"
    >
      Test
    </span>
  </div>
</div>
`;

exports[`Alert snapshot tests should render warning alert properly 1`] = `
<div
  className="alert alert--warning"
>
  <div
    className="alert__body"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-exclamation-triangle fa-w-18 alert__icon alert__icon--warning"
      data-icon="exclamation-triangle"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248.747 204.705l6.588 112c.373 6.343 5.626 11.295 11.979 11.295h41.37a12 12 0 0 0 11.979-11.295l6.588-112c.405-6.893-5.075-12.705-11.979-12.705h-54.547c-6.903 0-12.383 5.812-11.978 12.705zM330 384c0 23.196-18.804 42-42 42s-42-18.804-42-42 18.804-42 42-42 42 18.804 42 42zm-.423-360.015c-18.433-31.951-64.687-32.009-83.154 0L6.477 440.013C-11.945 471.946 11.118 512 48.054 512H527.94c36.865 0 60.035-39.993 41.577-71.987L329.577 23.985zM53.191 455.002L282.803 57.008c2.309-4.002 8.085-4.002 10.394 0l229.612 397.993c2.308 4-.579 8.998-5.197 8.998H58.388c-4.617.001-7.504-4.997-5.197-8.997z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="alert__content"
    >
      Test
    </span>
  </div>
</div>
`;

exports[`Alert snapshot tests should render with className 1`] = `
<div
  className="alert alert--success test"
>
  <div
    className="alert__body"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-check-circle fa-w-16 alert__icon alert__icon--success"
      data-icon="check-circle"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="alert__content"
    >
      Test
    </span>
  </div>
</div>
`;

exports[`Alert snapshot tests should render with close button when showClose is true 1`] = `
<div
  className="alert alert--success"
>
  <div
    className="alert__body"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-check-circle fa-w-16 alert__icon alert__icon--success"
      data-icon="check-circle"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="alert__content"
    >
      Test
    </span>
  </div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-times fa-w-10 alert__close"
    data-icon="times"
    data-prefix="far"
    focusable="false"
    onClick={[Function]}
    role="img"
    style={Object {}}
    viewBox="0 0 320 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;
